flink-runtime/src/test/java/org/apache/flink/runtime/concurrent/FutureUtilsTest.java

```diff
@@ -235,6 +237,57 @@ public void testRetryWithDelayCancellation() {
 		assertTrue(scheduledFuture.isCancelled());
 	}
 
+	/**
+	 * Tests that the operation could be scheduled with expected delay.
+	 */
+	@Test
+	public void testScheduleAsyncDelay() throws Exception  {
+		final Time delay = Time.milliseconds(500L);
+		final long start = System.currentTimeMillis();
+		final AtomicInteger updateValue = new AtomicInteger(0);
+
+		CompletableFuture<Integer> scheduleAsyncFuture = FutureUtils.scheduleAsync(
+			() -> updateValue.addAndGet(ThreadLocalRandom.current().nextInt(100)),
+			delay,
+			TestingUtils.defaultScheduledExecutor());
+
+		int result = scheduleAsyncFuture.get();
+		long completionTime = System.currentTimeMillis() - start;
```

Comment:
There some issues that I see here:
1. `System.currentTimeMillis()` is [impacted by system clock changes](https://stackoverflow.com/a/15620527) – this may cuase test failures in conjunction with NTP. Therefore `System.nanoTime()` should be preferred.  
1. Test will run at least 500ms which is on the slow side for unit tests. I think it's enough to use `ManuallyTriggeredScheduledExecutor` and see if the correct result is produced.
1. I don't think using `AtomicInteger` adds test coverage. Imo it's enough to assert on the value of `result`.